### DISCUSSION TITLE
p/9751355186fd4c5bbad0d041fe85c256/

### DISCUSSION THREAD

**Discussion by giscus** [rxliuli/blog#57] [Announcements]
# p/9751355186fd4c5bbad0d041fe85c256/

前言最近两个月一直在看 SICP 的 Youtube 课程，作为一个半路出家的人，自觉基础欠缺，至今已是工作的第四个年头，正如年初计划所言，仅仅依赖于热情去折腾各种简单的玩具已经不再能有效地提升自己了，需要系统化的学习和组织知识。虽然之前也买过实体书，但由于看起来比较难而且也没有风趣的讲解（看的有点枯燥，看完视频后感觉里面的教授真的太有趣了！），所以一直没能看下去（另外一本《算法》也是同样的命运）

https://blog.rxliuli.com/p/9751355186fd4c5bbad0d041fe85c256/

**Comment by selfboot** [comment]
你这个实现太复杂了，可以参考我的哈：

https://gallery.selfboot.cn/en/algorithms/hanoitower

这里是可视化交互，支持手动和自动模式来玩汉诺塔，支持从任意初始状态开始。

这里交互的话，获取提示就是从任意状态开始找到下一步移动方法的。算法可以看代码，开源的哈
